File: src/main/java/io/github/asyncronous/exo/handler/EXOToolTipHandler.java
package io.github.asyncronous.exo.handler;

import cpw.mods.fml.common.eventhandler.SubscribeEvent;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraftforge.event.entity.player.ItemTooltipEvent;

import io.github.asyncronous.exo.EXOTag;
import io.github.asyncronous.exo.item.ItemExoskeletonArmor;

public final class EXOToolTipHandler{
    @SubscribeEvent
    public void onToolTip(ItemTooltipEvent e){
        if(e.itemStack.getItem() instanceof ItemExoskeletonArmor){
            NBTTagCompound comp = EXOTag.getTag(e.itemStack);
            e.toolTip.add("Core: " + comp.getString("coreName"));
        }
    }
}

File: src/main/scala/io/github/asyncronous/exoskeleton/handler/ExoToolTipHandler.scala
package io.github.asyncronous.exoskeleton.handler

import cpw.mods.fml.common.eventhandler.SubscribeEvent
import io.github.asyncronous.exoskeleton.ExoskeletonTag
import io.github.asyncronous.exoskeleton.item.ItemExoArmor
import net.minecraft.nbt.NBTTagCompound
import net.minecraftforge.event.entity.player.ItemTooltipEvent

object ExoToolTipHandler{
  @SubscribeEvent
  def onToolTip(e: ItemTooltipEvent): Unit ={
    if(e.itemStack.hasTagCompound &&
       e.itemStack.getTagCompound.hasKey(ExoskeletonTag.IDENTIFIER) &&
       e.itemStack.getItem.isInstanceOf[ItemExoArmor]){

      val comp: NBTTagCompound = ExoskeletonTag.getTag(e.itemStack);
      e.toolTip.add("Current Core: " + comp.getString("coreName"));
    }
  }
}